internal/input/input.go

```diff
@@ -0,0 +1,149 @@
+package input
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/ap-pauloafonso/ratio-spoof/internal/bencode"
+)
+
+type InputArgs struct {
+	TorrentPath       string
+	InitialDownloaded string
+	DownloadSpeed     string
+	InitialUploaded   string
+	UploadSpeed       string
+	Port              int
+	Debug             bool
+}
+
+type InputParsed struct {
+	TorrentPath       string
+	InitialDownloaded int
+	DownloadSpeed     int
+	InitialUploaded   int
+	UploadSpeed       int
+	Port              int
+	Debug             bool
+}
+
+var validInitialSufixes = [...]string{"%", "b", "kb", "mb", "gb", "tb"}
+var validSpeedSufixes = [...]string{"kbps", "mbps"}
+
+func (I *InputArgs) ParseInput(torrentInfo *bencode.TorrentInfo) (*InputParsed, error) {
+	downloaded, err := extractInputInitialByteCount(I.InitialDownloaded, torrentInfo.TotalSize, true)
+	if err != nil {
+		return nil, err
+	}
+	uploaded, err := extractInputInitialByteCount(I.InitialUploaded, torrentInfo.TotalSize, false)
+	if err != nil {
+		return nil, err
+	}
+	downloadSpeed, err := extractInputByteSpeed(I.DownloadSpeed)
+	if err != nil {
+		return nil, err
+	}
+	uploadSpeed, err := extractInputByteSpeed(I.UploadSpeed)
+	if err != nil {
+		return nil, err
+	}
+
+	if I.Port < 1 || I.Port > 65535 {
```

Comment:
To avoid magic numbers, I would recommend assigning 1 and 65535 to meaningful variables or constants.

Reply:
Done, i created these two constants **minPortNumber** and **maxPortNumber**